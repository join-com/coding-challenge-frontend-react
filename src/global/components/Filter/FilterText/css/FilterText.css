.FilterText__label {
  display: block;
  font-size: 1.5rem;
  font-weight: bold;
}

.FilterText__input {
  margin-top: calc(var(--space) / 3);
  display: block;
}

.FilterText__input[placeholder] {
  font-weight: normal;
}
