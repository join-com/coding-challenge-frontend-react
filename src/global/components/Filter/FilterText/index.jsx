import React from 'react';

import PropTypes from 'prop-types';
import filterType from '../types/filter';

import { bemCls } from '../../../utils/ClassNameHelpers';

import InputText from '../../Inputs/InputText';

import './css/FilterText.css';

const FilterText = ({
  filter, value, className, handleFiltersChange,
}) => {
  if (filter) {
    const {
      name, type, label, placeholder,
    } = filter;

    return (
      <div className={`FilterText ${className}`}>
        <InputText
          className={`FilterText ${className} ${
            bemCls(`FilterText ${className}`, `__${name}`)}`}
          name={name}
          type={type}
          label={label}
          placeholder={placeholder}
          value={value}
          onChange={handleFiltersChange}
        />
      </div>
    );
  }
  return null;
};

export default FilterText;

FilterText.propTypes = {
  filter: filterType,
  value: PropTypes.string,
  className: PropTypes.string,
  handleFiltersChange: PropTypes.func,
};

FilterText.defaultProps = {
  filter: null,
  value: '',
  className: '',
  handleFiltersChange: () => {},
};
