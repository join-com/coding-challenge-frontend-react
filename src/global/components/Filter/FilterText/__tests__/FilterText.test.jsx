import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FilterText from '../index';

const props = {
  filter: {
    label: 'MyFilterInputLAbel',
    name: 'myFilterInputName',
    placeholder: 'My Filter Input Placeholder',
    type: 'text',
  },
  value: 'MyFilterValue',
  className: 'MyFilterClassname',
  handleFiltersChange: () => {},
};

describe('FilterText', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<FilterText {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
