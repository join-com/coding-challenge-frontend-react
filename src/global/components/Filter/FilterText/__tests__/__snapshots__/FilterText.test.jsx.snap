// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterText should render the component correctly 1`] = `
<div
  className="FilterText MyFilterClassname"
>
  <InputTextWrapper
    className="FilterText MyFilterClassname FilterText__myFilterInputName MyFilterClassname__myFilterInputName"
    label="MyFilterInputLAbel"
    name="myFilterInputName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="My Filter Input Placeholder"
    type="text"
    value="MyFilterValue"
  />
</div>
`;
